src/arithmetic/algebraic_expression/algebraic_expression_optimization.c

```diff
@@ -328,17 +339,19 @@ static void _Pushdown_TransposeExp
 		_Pushdown_TransposeOperation(exp);
 		break;
 	case AL_OPERAND:
-		_Pushdown_TransposeOperand(exp);
+		_AlgebraicExpression_TransposeOperand(exp);
```

Comment:
I don't think PushDown should perform transpose,
this function should simply relocate nodes within the tree.

there should be another optimisation function following pushdown which replaces transpose operations acting directly on operands with just the transposed operand.